react/div-color-palette/src/main.js

```diff
@@ -0,0 +1,35 @@
+import React, {useState} from 'react';
+import ReactDOM from 'react-dom';
+import ColorDiv from './colorDiv';
+
+const App = () => {
+  const [color, setColor] = useState(null);
+  const items = []
+
+  let tinycolor = require("tinycolor2");
+  console.log("colorPicker", color);
+
+  function getNextColor(source, offset) {
+    return tinycolor(source).lighten(offset).toHexString();
+  }
+
+  let offset = 50;
+  for (let i = 0; i < 10; i++) {
```

Comment:
When doing iteration in react you will almost always see the use of arrays, since you actually change nothing between iterations you can simply do this:
```js
{Array(10).fill().map((item, index) => 
  <ColorDiv key={i} color={nextColor}/>)
}
```

Reply:
I agree that this looks cleaner, but I doesn't contain the offset increment and not the call to the get the nextColor. I tried to add those lines but looks like that you can only use this to build the div :-(

Reply:
@moshesrur don't copy-paste, this is not the complete solution, just an example;)